src/cmdstan/command.hpp

```diff
@@ -163,7 +163,7 @@ int command(int argc, const char *argv[]) {
         list_argument *metric
             = dynamic_cast<list_argument *>(algo->arg("hmc")->arg("metric"));
         if (engine->value() != "nuts"
-            && (metric->value() != "dense_e" || metric->value() == "diag_e")) {
+            && !(metric->value() == "dense_e" || metric->value() == "diag_e")) {
```

Comment:
Was this just a bug before? This change changes the logic of the conditional but the new version seems like what was actually intended all along?

Reply:
I'm honestly not sure. 

To me, this looks like a kind of belt and suspsenders check.
Changing the logic didn't break the unit tests, which just means that this isn't actually tested.
There are two sampling algorithms:  hmc and fixed params, where "fixed_params" is a model without parameters,
and therefore no (or and empty) metric.

My call was to make the code agree with the comment, but that's being lazy.   The proper thing to do is clean up the logic, but I'm just not sure it's worth it.

Reply:
First refactor the conditional by moving the operators inside the disjunction

```cpp
engine->value() != "nuts"
&& metric->value() != "dense_e"
&& metric->value() != "diag_e"
```

I don't think this is NUTS-specific, so I'd remove the first line.

Then the remaining conjuncts with

```cpp
metric->value() == "unit_e"
```

because that's the metric that does not support adaptation.

One might argue that the negated form is better because it explicitly negates the things that do have adaptation.  I'd also be OK with that reasoning, but I still think you can remove the `!= "nuts"` test.

Reply:
> I still think you can remove the != "nuts" test.

The num_chains argument is also unsupported for `engine=static`, so somewhere it does need to check that it is NUTS

Reply:
refactored as helper function `validate_multi_chain_config`

working with the config for the sampler is a complete nightmare.  any access of a subarg which isn't actualy present will segfault.   to bullet-proof this would require more refactoring that it's worth, as there aren't that many folks using the command line directoy, and the CmdStanX wrapper interfaces vet the config beforehand.

Reply:
Is there a way to test if a subarg exists?

Reply:
The `categorical_argument` differs from the `list_argument` in that it doesn't have an internal cursor which points into the list of subarguments.   I'm just not good enough at type/template hacking to see how to deal with this.
Bottom line is that dealing with the config of NUTS requires tedious traversal of the argument tree.